### DISCUSSION TITLE
401 Error for Spring Boot resource server when deployed. Works fine in local dev environment

### DISCUSSION THREAD

**Discussion by manga1a** [zitadel/zitadel#5803] [Q&A] [answered]
I am using JWT bearer token to access my Spring boot gateway resource server. Authorization works fine running in localhost (both IDE and docker compose). But when I test with the same container image hosted, I get a 401 Unauthorized error. Client credentials settings used for local testing and hosted are exactly the same. Response has the following error message:
```
Bearer error="invalid_token", error_description="Failed to validate the token", error_uri="https://tools.ietf.org/html/rfc6750#section-3.1"
```
Following is my security config:
```
@EnableWebFluxSecurity
public class SecurityConfig {

    @Bean
    SecurityWebFilterChain springSecurityFilterChain(ServerHttpSecurity http) {
        http
                .csrf().disable()
                .authorizeExchange(exchanges -> exchanges
                                .anyExchange().authenticated()
                )
                .oauth2ResourceServer(ServerHttpSecurity.OAuth2ResourceServerSpec::jwt);

        return http.build();
    }
}
```
I set the issuer-uri as following:
```
spring:
  security:
    oauth2:
      resourceserver:
        jwt:
          issuer-uri: "https://agilesoft-ofs-xvsmie.zitadel.cloud"
```
Following is the debug log leading up to the authorization failure:
```
o.s.w.r.f.client.ExchangeFunctions       : [6af2a84f] HTTP GET https://agilesoft-test.au.auth0.com/.well-known/jwks.json
o.s.w.r.f.client.ExchangeFunctions       : [6af2a84f] [4b8fa1dd-1] Response 200 OK
o.s.w.r.f.client.ExchangeFunctions       : [68bedcdd] HTTP GET https://agilesoft-test.au.auth0.com/.well-known/jwks.json
o.s.w.r.f.client.ExchangeFunctions       : [68bedcdd] [3c501bda-1] Response 200 OK
o.s.s.w.s.a.AuthenticationWebFilter      : Authentication failed: Failed to validate the token
o.s.w.r.f.client.ExchangeFunctions       : [6615841f] HTTP GET https://agilesoft-test.au.auth0.com/.well-known/jwks.json
o.s.w.r.f.client.ExchangeFunctions       : [6615841f] [4b8fa1dd-2] Response 200 OK
o.s.s.w.s.a.AuthenticationWebFilter      : Authentication failed: Failed to validate the token
```

Appreciate any clue in resolving this. Thank you.

**Comment by mffap** [comment, marked as answer]
> Following is the debug log leading up to the authorization failure:
> 
> ```
> o.s.w.r.f.client.ExchangeFunctions       : [6af2a84f] HTTP GET https://agilesoft-test.au.auth0.com/.well-known/jwks.json
> o.s.w.r.f.client.ExchangeFunctions       : [6af2a84f] [4b8fa1dd-1] Response 200 OK
> o.s.w.r.f.client.ExchangeFunctions       : [68bedcdd] HTTP GET https://agilesoft-test.au.auth0.com/.well-known/jwks.json
> o.s.w.r.f.client.ExchangeFunctions       : [68bedcdd] [3c501bda-1] Response 200 OK
> o.s.s.w.s.a.AuthenticationWebFilter      : Authentication failed: Failed to validate the token
> o.s.w.r.f.client.ExchangeFunctions       : [6615841f] HTTP GET https://agilesoft-test.au.auth0.com/.well-known/jwks.json
> o.s.w.r.f.client.ExchangeFunctions       : [6615841f] [4b8fa1dd-2] Response 200 OK
> o.s.s.w.s.a.AuthenticationWebFilter      : Authentication failed: Failed to validate the token
> ```

Have you configured an auth0 tenant in the hosted environment? Seems like these requests are directed to auth0.com

**Reply by manga1a** [reply]
Thank you very much for the quick response. Just realized my mistake after pasting the log :-/. Yes, I had overriding issuer-uri in K8S deployment.
